### PR TITLE
EUI-6813 CAA filter no cases found message

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-6813

### Change description ###
CAA assigned cases - No cases found message for Assigned and Unassigned cases filter  

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by tarikdogan-hmcts — APPROVED** [review]

**Review by maliumer — APPROVED** [review]

File: api/caaCaseTypes/caaCaseTypes.util.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
 import { CaaCasesPageType } from '../caaCases/enums';
 import { searchCasesString } from './caaCaseTypes.constants';
 
-export function getRequestBody(organisationID: string, caaCasesPageType: string) {
+export function getRequestBody(organisationID: string, caaCasesPageType: string, caaCasesFilterValue?: string | string[]) {
   const organisationAssignedUsersKey = `supplementary_data.orgs_assigned_users.${organisationID}`;
+  const reference = 'reference.keyword';
+  let caseReferenceFilter: any[] = [];
```
Comment (Daniel-Lam) [inline api/caaCaseTypes/caaCaseTypes.util.ts:7]: @johnbenjamin-hmcts - `caseReferenceFilter` can be a `const`, if it's not being reassigned

File: api/caaCaseTypes/caaCaseTypes.util.ts [hunk 2]
```diff
@@ -28,11 +40,17 @@ export function getRequestBody(organisationID: string, caaCasesPageType: string)
                 must_not: [
                   { range: { [organisationAssignedUsersKey]: { gt: 0 } } }
                 ]
+              })
+            }
+          },
+          {
+            bool: {
+              ...(caaCasesFilterValue && !Array.isArray(caaCasesFilterValue) && {
+                must: caseReferenceFilter
```
Comment (Daniel-Lam) [inline api/caaCaseTypes/caaCaseTypes.util.ts:49]: @johnbenjamin-hmcts - just checking that the use of `must` on Line 49 and `should` on Line 52 is intentional

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts [hunk 3]
```diff
@@ -44,54 +45,102 @@ export class CaaCasesComponent implements OnInit {
...
+
+    // Load users of selected organisation from store
+    this.userStore.dispatch(new fromUserStore.LoadUsers(0));
+    this.selectedOrganisationUsers$ = this.userStore.pipe(select(fromUserStore.getGetUserList));
+
+    // Load cases based on page type and set table configuration
+    this.loadCasesAndSetTableConfig();
+  }
+
+  public loadCaseTypes(selectedFilterType: string, selectedFilterValue: string): void {
```
Comment (Daniel-Lam) [inline src/caa-cases/containers/caa-cases/caa-cases.component.ts:85]: @johnbenjamin-hmcts - I may have missed it but is there a unit test for this function, or is it covered indirectly?

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts [hunk 4]
```diff
@@ -145,29 +176,31 @@ export class CaaCasesComponent implements OnInit {
...
-    } else {
-      this.store.dispatch(new fromStore.LoadAssignedCases({
-        caseType: this.currentCaseType,
-        pageNo: this.currentPageNo,
-        pageSize: this.paginationPageSize,
-        caaCasesFilterType: this.selectedFilterType,
-        caaCasesFilterValue: this.selectedFilterValue
-      }));
-      this.cases$ = this.store.pipe(select(fromStore.getAllAssignedCaseData));
+    if (this.currentCaseType) {
```
Comment (Daniel-Lam) [inline src/caa-cases/containers/caa-cases/caa-cases.component.ts:179]: @johnbenjamin-hmcts - would be good to have this covered by a unit test, even though there wasn't one previously. Something basic, just to check if actions are being dispatched when they should, and not when they shouldn't

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just a few observations and some suggestions for unit tests

File: api/caaCaseTypes/caaCaseTypes.util.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply api/caaCaseTypes/caaCaseTypes.util.ts:49]: @Daniel-Lam - we have implemented the same logic when searching for cases. The filter keyword 'must' is when searching for a single case reference and 'should' is when searching for multiple case references.

File: api/caaCaseTypes/caaCaseTypes.util.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply api/caaCaseTypes/caaCaseTypes.util.ts]: @Daniel-Lam - I have now updated the code.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:85]: Thank you @Daniel-Lam I have now written couple of unit tests to cover this.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:179]: Thank you @Daniel-Lam I have now written couple of unit tests to cover this.

Comment (Daniel-Lam) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:179]: Thanks for adding some unit tests, @johnbenjamin-hmcts.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 3 above)
Comment (Daniel-Lam) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:85]: Thanks for adding some unit tests, @johnbenjamin-hmcts.

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**